crates/cext/src/transpiler/passes/commutative_cancellation.rs

```diff
@@ -0,0 +1,124 @@
+// This code is part of Qiskit.
+//
+// (C) Copyright IBM 2025
+//
+// This code is licensed under the Apache License, Version 2.0. You may
+// obtain a copy of this license in the LICENSE.txt file in the root directory
+// of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+//
+// Any modifications or derivative works of this code must retain this
+// copyright notice, and modified files need to carry a notice indicating
+// that they have been altered from the originals.
+
+use crate::pointers::{const_ptr_as_ref, mut_ptr_as_ref};
+
+use qiskit_circuit::circuit_data::CircuitData;
+use qiskit_circuit::converters::dag_to_circuit;
+use qiskit_circuit::dag_circuit::DAGCircuit;
+use qiskit_transpiler::commutation_checker::get_standard_commutation_checker;
+use qiskit_transpiler::passes::cancel_commutations;
+use qiskit_transpiler::target::Target;
+
+/// @ingroup QkTranspilerPasses
+/// Run the CommutativeCancellation transpiler pass on a circuit.
+///
+/// This pass cancels the redundant (self-adjoint) gates through commutation relations.
+///
+/// @param circuit A pointer to the circuit to run CommutativeCancellation on. This circuit
+/// pointed to will be updated with the modified circuit if the pass is able to remove any gates.
+/// @param target The target for the CommutativeCancellation pass. The target is used to specify
+/// as fallback Z rotation gate to use for cancellations if one can't be identified in the circuit.
+/// This can be a null pointer which indicates there isn't a target to use.
+///
+/// @returns The integer return code where 0 represents no error and 1 is
+/// used to indicate an error was encountered during the execution of the pass.
+///
+/// # Example
+///
+/// ```c
+///     QkCircuit *qc = qk_circuit_new(4, 0);
+///     uint_t cx_qargs[2] = {0, 1};
+///     qk_circuit_gate(qc, QkGate_CX, cx_qargs, NULL);
+///     qk_circuit_gate(qc, QkGate_Z, &0, NULL);
```

Comment:
`&0` is legit? I learned something new 🙂

Reply:
Fixed in: https://github.com/Qiskit/qiskit/pull/14831/commits/9467e5cd85e922045f1c22cd4e6923f5cb890659